src/ILCompiler/src/Program.cs

```diff
@@ -284,14 +288,24 @@ private int Run(string[] args)
                         entrypointModule = module;
                     }
 
-                    compilationRoots.Add(new ExportedMethodsRootProvider(module));
+                    if (!_isWasmCodegen)
+                    {
+                        compilationRoots.Add(new ExportedMethodsRootProvider(module));
+                    }
                 }
 
                 if (entrypointModule != null)
                 {
-                    LibraryInitializers libraryInitializers =
-                        new LibraryInitializers(typeSystemContext, _isCppCodegen);
-                    compilationRoots.Add(new MainMethodRootProvider(entrypointModule, libraryInitializers.LibraryInitializerMethods));
+                    if (!_isWasmCodegen)
+                    {
+                        LibraryInitializers libraryInitializers =
+                            new LibraryInitializers(typeSystemContext, _isCppCodegen, _isWasmCodegen);
+                        compilationRoots.Add(new MainMethodRootProvider(entrypointModule, libraryInitializers.LibraryInitializerMethods));
+                    }
+                    else
+                    {
+                        compilationRoots.Add(new RawMainMethodRootProvider(entrypointModule));
```

Comment:
There should be a TODO saying we're only doing this because wasm can't do anything.

Reply:
Done